tests/test_minimizer.py

```diff
@@ -1,33 +1,42 @@
 import pytest
 import numpy as np
+import pandas as pd
+from sklearn.compose import ColumnTransformer
 
 from sklearn.datasets import load_boston
+from sklearn.impute import SimpleImputer
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import OneHotEncoder, StandardScaler
 
 from apt.minimization import GeneralizeToRepresentative
 from sklearn.tree import DecisionTreeClassifier
+from apt.utils import get_iris_dataset, get_adult_dataset, get_nursery_dataset
 
 
 @pytest.fixture
 def data():
     return load_boston(return_X_y=True)
 
+
 def test_minimizer_params(data):
     # Assume two features, age and height, and boolean label
     cells = [{"id": 1, "ranges": {"age": {"start": None, "end": 38}, "height": {"start": None, "end": 170}}, "label": 0,
-              "representative": {"age": 26, "height": 149}},
+              'categories': {}, "representative": {"age": 26, "height": 149}},
              {"id": 2, "ranges": {"age": {"start": 39, "end": None}, "height": {"start": None, "end": 170}}, "label": 1,
-              "representative": {"age": 58, "height": 163}},
+              'categories': {}, "representative": {"age": 58, "height": 163}},
              {"id": 3, "ranges": {"age": {"start": None, "end": 38}, "height": {"start": 171, "end": None}}, "label": 0,
-              "representative": {"age": 31, "height": 184}},
+              'categories': {}, "representative": {"age": 31, "height": 184}},
              {"id": 4, "ranges": {"age": {"start": 39, "end": None}, "height": {"start": 171, "end": None}}, "label": 1,
-              "representative": {"age": 45, "height": 176}}
+              'categories': {}, "representative": {"age": 45, "height": 176}}
              ]
     features = ['age', 'height']
     X = np.array([[23, 165],
-                     [45, 158],
-                     [18, 190]])
+                  [45, 158],
+                  [18, 190]])
```

Comment:
Please remove prints from tests. We should replace with asserts where possible.

Reply:
what asserts can I do on minimizer

Reply:
The whole process is supposed to be deterministic, so you can copy the resulting generalizations and compare to them, for example.

Reply:
This is great. Can you also add at least one assert for the transformed data (for example that untouched features are not changed, or that features that receive range [] all have the same value)?

Reply:
Sorry to keep bugging you about this. The asserts you added are great. I just want one more to test that the data actually is generalized (when it's supposed to). Maybe the easiest way to do this is looking at the ncp scores (make sure that the transformed data has ncp>0). Or, we can compare the columns themselves (same as you did for the non-transformed features, only this time check **not** equals). Thanks!